docs/usage/contrib/sqlalchemy.rst

```diff
@@ -0,0 +1,35 @@
+SQLAlchemy Models & Repository
+==============================
+
+Litestar comes with a built-in repository class (`SQLAlchemyAsyncRepository`) for `SQLAlchemy <https://docs.sqlalchemy.org/>`_ to make CRUD operations easier.
+
+Features
+--------
+
+* Pre-configured `DeclarativeBase` for `SQLAlchemy <https://docs.sqlalchemy.org/>`_ 2.0 that includes a UUID based primary-key and an optional version with audit columns.
+* Generic asynchronous repository for select, insert, update, and delete operations on SQLAlchemy models
+* Implements optimized methods for bulk inserts, updates, and deletes.
+* Integrated counts, pagination, filtering with `LIKE`, `IN`, and dates before and/or after.
+* Support for SQLite via `aiosqlite` and Postgres via `asyncpg`
+
+Basic Use
+---------
+
+To use the `SQLAlchemyAsyncRepository` repository, you must first define your models using one of the included built-in `DeclarativeBase` ORM base implementations  (`Base` and `AuditBase`).  Both include a UUID based primary key and `AuditBase` includes an `updated` and `created` timestamp column.
+
+Models using these bases also include the following enhancements:
+* Auto-generated snake-case table name from class name
+* Pydantic BaseModel and Dict classes map to JSONB SQLAlchemy type map
```

Comment:
```suggestion
* Pydantic BaseModel and Dict classes map to :class:`JSONB <sqlalchemy.dialects.postgresql.JSONB>` SQLAlchemy type map
```

Not sure if that's the correct path for the reference

Reply:
I think it should be :class: `sqlalchemy.types.JSON` this is the generic implementation that automatically selects the dialect specific version.

Reply:
Afaik for SQLA itself, `JSON` does not render as `JSONB` for postgres since postgres has both. Does the repo do that? If so, that should probably be mentioned (and is a sensible default).

Reply:
well, it looks like you are right.  I was thinking that it would use the `JSONB` implementation for Postgres.  Maybe I should add a custom SQLAlchemy JSON type that handles this for dialects that support JSONB.